### PR TITLE
docs: Improve Developer-Guide and SRIOV doc on arm64

### PR DESCRIPTION
**Pull request by hejianet**
This improves the guest kernel build section of Developer-Guide
on arm64.
This also improve the description of sriov use-case.

Fixes: #299
Signed-off-by: Jia He <justin.he@arm.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Thanks @hejianet!

lgtm

**Comment by jodh-intel** [comment]
Ping @kata-containers/documentation!

File: Developer-Guide.md [hunk 1]
```diff
@@ -342,7 +342,7 @@ $ curl -L https://raw.githubusercontent.com/kata-containers/packaging/master/ker
 $ make ARCH=${kernel_dir} -j$(nproc)
 $ kata_kernel_dir="/usr/share/kata-containers"
 $ kata_vmlinuz="${kata_kernel_dir}/kata-vmlinuz-${kernel_version}.container"
-$ [ $kernel_arch = ppc64le ] && kernel_file="$(realpath ./vmlinux)" || kernel_file="$(realpath arch/${kernel_arch}/boot/bzImage)"
+$ [ $kernel_arch = ppc64le ] && kernel_file="$(realpath ./vmlinux)" || [ $kernel_arch = aarch64 ] && kernel_file="$(realpath arch/arm64/boot/Image)" || kernel_file="$(realpath arch/${kernel_arch}/boot/bzImage)"
```
Comment (marcov) [inline Developer-Guide.md]: This is not working as intended, you need a proper `if..elfi...fi` or `case...esac` like:

```
$ case $kernel_arch in; ppc64le) kernel_path="./vmlinux";; aarch64) kernel_path="arch/arm64/boot/Image";; *) kernel_path="arch/${kernel_arch}/boot/bzImage";; esac
$ kernel_file="$(realpath $kernel_path)"
```

File: use-cases/using-SRIOV-and-kata.md [hunk 2]
```diff
@@ -41,6 +41,10 @@ In order to setup your host for SR-IOV, the following has to be true:
   Controller (NIC). The section [Checking your NIC for SR-IOV](#checking-your-nic-for-sr-iov) describes how to assess if you need to make NIC changes and how to make
   the necessary changes.
 
+Besides, you need to enable the NIC driver in your guest kernel config:
+eg. mlx5 for Mellanox NIC.
```
Comment (marcov) [inline use-cases/using-SRIOV-and-kata.md]: I'd rewrite it as following:

Besides, you need to enable the NIC driver in your guest kernel config (e.g. `mlx5` for Mellanox NIC).

File: Developer-Guide.md (hunk 1 above)
Comment (hejianet) [reply Developer-Guide.md]: thanks, updated

File: use-cases/using-SRIOV-and-kata.md (hunk 2 above)
Comment (hejianet) [reply use-cases/using-SRIOV-and-kata.md]: thanks, updated

File: Developer-Guide.md (hunk 1 above)
Comment (marcov) [reply Developer-Guide.md]: Thank you!

**Comment by marcov** [comment]
lgtm

**Pull request merged**